### PR TITLE
✨ Created decorator to include viadot source to df

### PR DESCRIPTION
**Pull request by fgoiriz**
## Summary

"In this PR a decorator that adds a new column indicating the source name is added to every source to_df method"

## Importance

Because we need to register inside the df the source data is coming from

## Checklist

This PR:

- [x] adds new tests (if appropriate)
- [x] updates docstrings for any new functions or function arguments (if appropriate)
- [ ] updates `CHANGELOG.md` with a summary of the changes

### CODE REVIEW

File: viadot/utils.py [hunk 1]
```diff
@@ -396,5 +397,25 @@ def call_shell(command):
...
+    df.columns = (
+        df.columns.str.strip().str.replace(" ", "_").str.replace("-", "_").str.lower()
+    )
     return df
+
+
+def add_viadot_source_column(func):
+    """Decorator to be added to 'to_df' methods of sources.
+    Adds new '_viadot_source' column to the resulting df
+    indicating the name of the souce the df is coming from"""
```
Comment (Trymzet) [inline viadot/utils.py:409]: I'd keep the quotes on separate lines, also you have a typo ("source").

File: tests/unit/test_utils.py [hunk 2]
```diff
@@ -66,3 +66,16 @@ def test_double_quotes_inside():
 
 VALUES ({TEST_VALUE_ESCAPED}, 'c')"""
     ), test_insert_query
+
+
+def test_add_viadot_source_column():
+    class Testing_class:
```
[the remark this answers is not in the document]
Comment (fgoiriz) [reply tests/unit/test_utils.py:72]: Noted, thanks.

File: viadot/utils.py (hunk 1 above)
Comment (fgoiriz) [reply viadot/utils.py:409]: Modified

File: viadot/sources/azure_data_lake.py [hunk 3]
```diff
@@ -4,7 +4,7 @@
 import pandas as pd
 from adlfs import AzureBlobFileSystem, AzureDatalakeFileSystem
 
-from ..config import get_source_credentials
+from ..config import get_source_credentials, add_viadot_source_column
```
[the remark this answers is not in the document]
Comment (fgoiriz) [reply viadot/sources/azure_data_lake.py:7]: Solved

**Review by Trymzet — APPROVED** [review]
Looks good, thanks for adding the missing typing as well :)

**Pull request merged**